snuba/consumer.py

```diff
@@ -475,6 +469,44 @@ def process_message_multistorage(
     return results
 
 
+class DevNullBatchWriter(ProcessingStep[JSONRowInsertBatch]):
+    def __init__(self, metrics: MetricsBackend) -> None:
+        self.__metrics = metrics
+
+        self.__messages: MutableSequence[Message[JSONRowInsertBatch]] = []
+        self.__closed = False
+
+    def poll(self) -> None:
+        pass
+
+    def submit(self, message: Message[JSONRowInsertBatch]) -> None:
+        assert not self.__closed
+
+        self.__messages.append(message)
+
+    def close(self) -> None:
+        self.__closed = True
+
+        if not self.__messages:
+            return
+
+        write_start = time.time()
+        time.sleep(2 + random.random() - 0.5)
```

Comment:
Why is this time random? Wouldn't it be easier to analyze if it was consistent?

Reply:
It is not random. It is 2 seconds with a random jitter of +/- 0.5 seconds. To simulate a realistic environment.